source/common/upstream/BUILD

```diff
@@ -159,6 +160,20 @@ envoy_cc_library(
     ],
 )
 
+envoy_cc_library(
```

Comment:
Now that you have the HDS skeleton working, I wonder if we should take some time to think about where it should ideally live. I think it really is a server global config and independent of `ClusterManager` logically, since it might be asked to health check hosts that don't even belong to a configured cluster on the Envoy instance. OTOH, `ClusterManager` has a lot of machinery for dealing with clusters, associated host sets and their health checks/status.

This goes back to the observation that when Envoy acts as a health checker delegate, it's acting essentially independent of its regular proxy duties. Maybe a second `ClusterManager` instantiation would make sense for HDS purposes that is never used for serving? @mattklein123 any thoughts on this organization?

Reply:
To answer my own question, @markatou and I chatted a bit and it seems that CM doesn't really provide us much value here, we should make the `HdsDelegate` owned by `Server::Instance` and it can instantiate some custom subclass of `Upstream::Cluster` to attach health checkers to. We don't need to manage connection pools, thread-local cluster status, etc.